util/ssh_client.go

```diff
@@ -29,6 +51,109 @@ func (c *sshClientImpl) ExecCommand(username string, password string, ip string,
 	return string(output), err
 }
 
+func (c *sshClientImpl) UploadFile(username string, password string, ip string, srcFile string, destFile string) error {
+	config := &myssh.ClientConfig{
+		User: username,
+		Auth: []myssh.AuthMethod{
+			myssh.Password(password),
+		},
+	}
+	if !IsIP(ip) {
+		return errors.New("invalid IP address")
+	}
+
+	if IsDir(srcFile) || IsDir(destFile) {
+		return errors.New("Is a directory")
+	}
+
+	client, err := myssh.Dial("tcp", ip+":22", config)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	sftp, err := sftp.NewClient(client)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	defer sftp.Close()
+
+	data, err := ioutil.ReadFile(srcFile)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	f, err := sftp.Create(destFile)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	_, err = f.Write([]byte(data))
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	_, err = sftp.Lstat(destFile)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	return nil
+}
+
+func (c *sshClientImpl) DownloadFile(username string, password string, ip string, srcFile string, destFile string) error {
+	config := &myssh.ClientConfig{
+		User: username,
+		Auth: []myssh.AuthMethod{
+			myssh.Password(password),
+		},
+	}
+
+	if !IsIP(ip) {
+		return errors.New("invalid IP address")
+	}
+
+	if IsDir(srcFile) || IsDir(destFile) {
+		return errors.New("Is a directory")
+	}
+
+	client, err := myssh.Dial("tcp", ip+":22", config)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	sftp, err := sftp.NewClient(client)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	defer sftp.Close()
+
+	pFile, err := sftp.Open(srcFile)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
```

Comment:
Please review all cases where a file is being open and make sure there is an error check and s defer to close the file. You missed one in previous file.